Backend/igc/IgcAnalyzer.js

```diff
@@ -28,15 +30,28 @@ const IgcAnalyzer = {
       igcWithReducedFixes,
       stripFactor
     );
-    writeStream.end(() => runOlc(pathToFile, flightId, stripFactor == null));
+    writeStream.end(() => runOlc(pathToFile, flightId, stripFactor == null, callback));
   },
+
+  extractFixes: (flight) =>{
+    //TODO Currently the file will be deserailized twice! 
+    //1x startCalculation and 1x extractFixes
+    const flightId = flight.id.toString();
+    const igcAsPlainText = readIgcFile(flightId);
+    const igcAsJson = IGCParser.parse(igcAsPlainText, { lenient: true });
+    reducedFixes = [];
+    for (i = 0; i < igcAsJson.fixes.length; i += IGC_FIXES_RESOLUTION) {
```

Comment:
This supposes that every igc logger uses a 1 sec interval which is not the case. This should be evaluated before reducing the fixes.